### DISCUSSION TITLE
Is there a delay in the availablity of API data products?

### DISCUSSION THREAD

**Discussion by victorthemtt** [ercot/api-specs#68] [Q&A] [answered]
Greetings, 

I'm trying to get the data via OpenAPI-specified endpoints in real time but I found there may be a large delay in the timeliness of certain products. For example, at 4:01 PM when I send a request for SCED_system_lembda, which is updated every 5 minutes, I can only get entries up to 3:50 PM or 3:55 PM, but never up to 4:00 PM. I wonder if this delay can be resolved. From my understanding, this delay shouldn't be present since I observed that it's updated quite timely on the product's website: https://www.ercot.com/mp/data-products/data-product-details?id=NP6-322-CD
Could someone share more information about this with me? Appreciate the help. 

Victor Mao

**Comment by ajay-ercot** [comment, marked as answer]
Victor, 

The data on Public Data Portal and API can be delayed for up to about 10-15 minutes max from when it is produced. The synchronization is currently done in a batch manner. However, we have project enhancements lined up for next year to make the data appear almost at the same time as it does on ercot.com. 

-Ajay, ERCOT

**Reply by victorthemtt** [reply]
Ajay, 

Thank you for your answer. Do you know any ways that I can get the most up to date data sooner? Web-scrapping ercot.com is one thing that comes to my mind but I wonder if better options are available. 

Victor Mao

**Reply by andrewa-habitat** [reply]
Hi @ajay-ercot if we are in need of data as soon as it's published, would you recommend using EWS instead of the Public API?

**Reply by ajay-ercot** [reply]
If you have a need to get access to the data products, as soon as they are available on ercot.com, then EWS SOAP API's or automation around the ercot.com HTTPS endpoints, will be the recommended option at this time. We have a release planned early in 2026 that will reduce the delay further to under 5 minutes. Later in 2026 we will have event driven architecture where the data will be available in the Public data portal/API at almost the same time as on ercot.com and EWS. Will report back on this forum as those changes are released.

**Reply by andrewa-habitat** [reply]
Thanks @ajay-ercot - by ercot.com HTTPS endpoints, are you referring to the download links found in the https://mis.ercot.com/misapp/servlets domains? thanks

**Reply by victorthemtt** [reply]
Hi @ajay-ercot, @andrewa-habitat, 

Thank you for the follow up. I assume the HTTPS endpoints you mentioned are https://mis.ercot.com/misapp/servlets. I'm using it to get latest data currently by adopting document ID lookups. It indeed has less delay than public data portal/API. Yet the shortcoming is it cannot get historical data many days ago. 
What's the EWS SOAP API? Any documents I can refer to? Thank you.
